app/views/items/_file.html.erb

```diff
@@ -0,0 +1,32 @@
+<div class="container"><%# Need additional container here %>
+  <div class="row">
+  Workspace:
+<% if @available_in_workspace &&  %>
+  <%= link_to params[:file], "/items/#{params[:id]}/file/?file=#{params[:file]}" -%>
+<% else %>
+  not available <span class="text-danger"><%= @available_in_workspace_error -%></span>
+<% end %>
+</div>
+
+<%# TODO: Use the contentMetadata OM for parsing metadata record %>
+<% @object.contentMetadata.ng_xml.search("//file[@id='#{params[:file]}']").each do |file|
+    if file['shelve'] == 'yes'
+ %>
+<div class="row">
+  Stacks:
+  <%= link_to params[:file], stacks_url_full_size(@object, params[:file]) -%>
+</div>
+<%
+    end
```

Comment:
I realize that this is legacy code porting over, though would you mind giving a separate block for the `<% end %>` and `if` statements?